packages/vite/src/node/config.ts

```diff
@@ -1450,10 +1453,20 @@ export async function resolveConfig(
         .getSortedPluginHooks('configResolved')
         .map((hook) => hook.call(resolvedConfigContext, workerResolved)),
     )
+    ;(workerResolved.plugins as Plugin[]) = resolvedWorkerPlugins
```

Comment:
Could we have a comment that explains why this need to mutate `workerResolved.plugins`? Also do we also need to mutate `workerResolved.environments`, or is that fine?

Reply:
Added comments, and rearranged the code so it is easier to follow (hopefully). Also avoided recreating the worker resolved config after passing it to hooks (as they may store a reference) and fixed resolvedConfig for hooks, that should be after resolve the environment plugins https://github.com/vitejs/vite/pull/20120/commits/ad08feeea707be1415075f5b58700b5a677b64b7